settings.gradle

```diff
@@ -52,37 +52,41 @@ project(':nessie').name = 'iceberg-nessie'
 project(':gcp').name = 'iceberg-gcp'
 project(':dell').name = 'iceberg-dell'
 
+List<String> knownScalaVersions = System.getProperty("knownScalaVersions").split(",")
+String scalaVersion = System.getProperty("scalaVersion", System.getProperty("defaultScalaVersion"))
+if (!knownScalaVersions.contains(scalaVersion)) {
+  throw new GradleException("Found unsupported Scala version: " + scalaVersion)
```

Comment:
Rather than throwing an exception, I think this should just exclude the projects from the build. Here's an example: https://github.com/apache/iceberg/blob/master/settings.gradle#L175-L193